src/main/java/org/apache/sysml/hops/FunctionOp.java

```diff
@@ -286,10 +322,17 @@ protected ExecType optFindExecType()
 		return _etype;
 	}
 
-	@Override
+	//@Override
 	public void refreshSizeInformation()
 	{
-		//do nothing
+		//only for refreshing the size of single output
+		//Hop inputx = null;
+
+		for(int i = 0; i < getInput().size(); i++) {
+			setDim1(getInput().get(i).getDim1());
+			setDim2(getInput().get(i).getDim2());
+		}
+
```

Comment:
this is dangerous because it sets the output to known size, which will be used by the memory estimates for the subsequent operation, but you cannot guarantee - without IPA - that the input sizes propagate to the output. Also the loop would be unnecessary, because the last input sizes override the output size anyway.